src/polymer/dom-module-scanner.ts

```diff
@@ -87,11 +93,20 @@ export class DomModuleScanner implements HtmlScanner {
 
     await visit((node) => {
       if (isDomModule(node)) {
+        const template = dom5.query(node, dom5.predicates.hasTagName('template'));
+        let slots: Slot[] = [];
+        if (template) {
+          slots = dom5.queryAll(
+            treeAdapters.default.getTemplateContent(template),
+            dom5.predicates.hasTagName('slot')
+          ).map(s => new Slot(dom5.getAttribute(s, 'name') || '', document.sourceRangeForNode(s)));
```

Comment:
The only way `document.sourceRangeForNode(s)` would return undefined here I think is if `s` wasn't originally from `document`, or if `document` wasn't parsed with location annotations turned on. We can be confident that neither was the case, so you can cast this to definitively non-null with `!` and make `slot.range` non optional as well